test/proposal_test.rb

```diff
@@ -232,5 +232,57 @@
         expect(sda6).to match_fields(mountpoint: "swap", uuid: "", label: "")
       end
     end
+
+    context "when installing on several GPT and MBR disks" do
+      let(:scenario) { "gpt_and_msdos" }
+      let(:separate_home) { true }
+      let(:lvm) { false }
+      let(:expected) do
+        file_name = "#{scenario}-#{yaml_suffix}"
+        ::Storage::Devicegraph.new_from_file(output_file_for(file_name))
+      end
+
+      before do
+        settings.candidate_devices = ["/dev/sda", "/dev/sdb"]
+        settings.root_device = root_device
+      end
+
+      context "if no disk is enforced for '/'" do
+        let(:root_device) { nil }
+        let(:yaml_suffix) { "no_root_device" }
+
+        include_examples "proposed layout"
+
+        it "uses the biggest suitable disk" do
+          proposal.propose
+          parts = proposal.devices.partitions
+          fs = parts.filesystems
+          expect(fs.with_mountpoint("/").disks.first.name).to eq "/dev/sdb"
+          expect(fs.with_mountpoint("/home").disks.first.name).to eq "/dev/sdb"
+          expect(fs.with_mountpoint("swap").disks.first.name).to eq "/dev/sdb"
+          expect(parts.with(id: Storage::ID_BIOS_BOOT).disks.first.name).to eq "/dev/sdb"
+        end
+      end
+
+      context "if a disk without free space is chosen for '/'" do
```

Comment:
I miss here complementary case what happen when there is a free space as I expect it will be more common.